framework/src/modules/chain/dpos/delegates_list.js

```diff
@@ -53,33 +53,54 @@ class DelegatesList extends EventEmitter {
 		return shuffleDelegateListForRound(round, list);
 	}
 
-	async getDelegatePublicKeysSortedByVote() {
+	async getDelegatePublicKeysSortedByVoteWeight(tx) {
 		const filters = { isDelegate: true };
 		const options = {
 			limit: this.activeDelegates,
 			sort: ['voteWeight:desc', 'publicKey:asc'],
 		};
-		const accounts = await this.storage.entities.Account.get(filters, options);
+		const accounts = await this.storage.entities.Account.get(
+			filters,
+			options,
+			tx,
+		);
 		return accounts.map(account => account.publicKey);
 	}
 
 	async generateActiveDelegateList(round) {
```

Comment:
I think we should rename this function now. Or maybe we can move the whole logic inside `getRoundDelegates` and get rid of this since now it's pretty simple what it's doing?

Reply:
There is one function using this logic which relies on the unshuffled logic.
Renamed functions to:
- `getForgersForRound`
- `getActiveDelegatesForRound`